src/frontend/hooks/useResolvedLanguageTag.test.tsx

```diff
@@ -0,0 +1,361 @@
+import {type ReactNode} from 'react';
+import {act, renderHook} from '@testing-library/react-native';
+import {
+  createSelectedLocaleStore,
+  SelectedLocaleStore,
+  SelectedLocaleStoreProvider,
+  useSelectedLocaleActions,
+} from '../contexts/SelectedLocaleContext';
+import {useResolvedLanguageTag} from './useResolvedLanguageTag';
+import {Locale, useLocales} from 'expo-localization';
+
+function createWrapper(selectedLocaleStore: SelectedLocaleStore) {
+  return ({children}: {children: ReactNode}) => {
+    return (
+      <SelectedLocaleStoreProvider value={selectedLocaleStore}>
+        {children}
+      </SelectedLocaleStoreProvider>
+    );
+  };
+}
+
+// Override the global mock that's done in jest.setup.js
+jest.mock('expo-localization', () => {
+  return {
+    useLocales: jest.fn(),
+  };
+});
+
+// Not strictly necessary to create this but helpful to make the tests a bit more descriptive
+function setSystemPreferredLocales(locales: string[]) {
+  jest.mocked(useLocales).mockReturnValue(locales.map(mockLocale));
+}
```

Comment:
there's probably a better way of going about this by mocking at a deeper level in expo-localization, but i think this is good enough for now.